app/components/dex-create.jsx

```diff
@@ -12,7 +12,16 @@ export class DexCreate extends Component {
 
   constructor (props) {
     super(props);
-    this.state = { error: null };
+    this.state = { error: null, generation: '7', region: 'national' };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange (event) {
+    if (this.state.generation === '7' && this.state.region === 'alola') {
```

Comment:
instead looking at the current state, you should look at what we're changing the state to. itll make the logic easier to understand in the future

```js
onChange = (e) => {
  const generation = parseInt(e.target.value);

  if (generation === 6) {
    this.setState({ region: 'national' });
  }

  this.setState({ generation });
}
```

^ this _should_ be the same logic, but you should double check